src/entity.ts

```diff
@@ -134,7 +134,7 @@ export class Entity {
 		if (!this._context)
 			this._context = context;
 		// Ensure provided context waits on the existing context to be ready
-		else
+		else if (this._context !== context)
```

Comment:
Are there any implications on the `if (context !== null) { ...` code at the bottom of the method? Previously it would either set the context and then null it out, or it would wait on the given context, then null it out, so the nulling out was at worst redundant in the second case? I suppose the same is still true? Still, might be more clear to only wait and null out the context in the method call when its assigned?

Reply:
Fixed.